src/services/CalcFactor.jl

```diff
@@ -20,44 +20,31 @@ _getDimensionsPartial(fct::DFGFactor) = _getDimensionsPartial(_getCCW(fct))
 _getDimensionsPartial(fg::AbstractDFG, lbl::Symbol) = _getDimensionsPartial(getFactor(fg, lbl))
 
 
-"""
-    $SIGNATURES
-Get `.factormetadata` for each CPT in CCW for a specific factor in `fg`. 
-"""
-_getFMdThread(ccw::CommonConvWrapper, 
-              thrid::Int=Threads.threadid()) = ccw.cpt[thrid].factormetadata
-#
-_getFMdThread(fc::Union{GenericFunctionNodeData,DFGFactor}, 
-              thrid::Int=Threads.threadid()) = _getFMdThread(_getCCW(fc), thrid)
-#
-_getFMdThread(dfg::AbstractDFG,
-              lbl::Symbol,
-              thrid::Int=Threads.threadid()) = _getFMdThread(_getCCW(dfg, lbl), thrid)
-#
-
 
 
 
 # Helper function to construct CF from a CCW
 function CalcFactor(ccwl::CommonConvWrapper;
-                    factor = ccwl.usrfnc!, 
-                    metadata = _getFMdThread(ccwl), 
+                    factor = ccwl.usrfnc!,
                     _sampleIdx = 0, 
                     _measCount = length(ccwl.measurement),
                     _legacyMeas = ccwl.measurement, 
                     _legacyParams = ccwl.params, 
                     _allowThreads = true,
-                    cache = ccwl.dummyCache )
+                    cache = ccwl.dummyCache,
+                    fullvariables=ccwl.fullvariables,
+                    solvefor=ccwl.varidx)
```

Comment:
we are also going to have to double check this with the multihypo recipes.  Basically the ground rule is that a CalcFactor object represents one specific hypothesis.  If there are particles with different hypos, then each particle gets a different CalcFactor object.

From that, `.fullvariables` and `solvefor` should only refer to the one active hypothesis this CalcFactor object is working with.

Reply:
> should fullvariables be filtered by hypo

Yes it should be filtered by hypo

Reply:
Ok. I figured that much, but couldn’t quite see how it was done / can be done without first removing CPT.